File: jbback/src/main/java/org/zmalchunz/configurer/jbback/template/TemplateResource.java
package org.zmalchunz.configurer.jbback.template;

import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.*;
import org.zmalchunz.configurer.jbback.template.api.Template;
import org.zmalchunz.configurer.jbback.template.api.TemplateRequest;

import java.util.List;
import java.util.UUID;

@RestController
@RequestMapping("/template")
@AllArgsConstructor
public class TemplateResource {

    private final TemplateService templateService;

    @GetMapping
    @ResponseBody
    public List<Template> getTemplates() {
        return templateService.getTemplates();
    }

    @PostMapping
    @ResponseBody
    public UUID addTemplates(@RequestBody TemplateRequest newTemplateRequest) {
        return templateService.createTemplate(newTemplateRequest.toTemplate());
    }
}

File: backend/src/main/kotlin/org/zmalchunz/configurer/backend/resources/TemplateResource.kt
package org.zmalchunz.configurer.backend.resources

import org.springframework.web.bind.annotation.*
import org.zmalchunz.configurer.backend.model.Template
import org.zmalchunz.configurer.backend.repositories.TemplateRepository

@RestController
@RequestMapping("/template")
class TemplateResource(val templateRepository: TemplateRepository) {

    @GetMapping
    fun get(): List<Template> = templateRepository.findTemplates()

    @PostMapping
    fun post(@RequestBody template: Template): String {
        return templateRepository.saveTemplate(template)
    }
}